widget/tree.go

```diff
@@ -0,0 +1,751 @@
+package widget
+
+import (
+	"image/color"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/driver/desktop"
+	"fyne.io/fyne/internal/widget"
+	"fyne.io/fyne/storage"
+	"fyne.io/fyne/theme"
+)
+
+const (
+	indentation = 4 // Multiplier of theme.Padding() to indent children by. // TODO consider moving to theme.Indentation()
```

Comment:
Given the recent discussion about alignment I wonder if this should not be something separate but instead theme.IconSize+theme.Padding?

Reply:
Ack. This would tie the child indentation to what the branch contains (the arrow icon + padding)